src/test/erasure-code/TestErasureCodeShec_arguments.cc

```diff
@@ -86,12 +86,12 @@ void create_table_shec432() {
           continue;
         }
         if (std::popcount(avails) == 4) {
-	  auto a = to_array<std::initializer_list<int>>({
+          std::initializer_list<std::initializer_list<int>> a {
```

Comment:
`a` is already a `std::initializer_list<..>` before this change. i don't understand what's the difference between before and after this change. can you reproduce this issue locally? if yes, which libstdc++ version are you using?

Reply:
**The problem is the scope of anonymous variables.**

The to_array function takes **an anonymous variable** as an argument.
Although the memory of this anonymous parameter are allocated within t**he stack of the current function**, but its scope is only in **the to_array function and its subfunctions**. 
**After the to_array function returns, the memory for the anonymous argument will be invalidated.**

Anonymous parameter std::initializer_list<.. > memory is located in the stack of the current function.
The internal member pointers of variable a just point to the memory address located in the stack of the current function which allocated for the anonymous variable's members.
This is analogous to the current function referencing the local variable address of the to_array function.
This results in accessing memory that is out of scope.

From the ASAN exception, you can see that the address is in the middle of the stack of the current function, and does not go beyond the scope of the current function's ’stack.

From the disassembly of the simplified code you can see the initialization process of the anonymous parameter std::initializer_list<...> and the memory of the anonymous parameter std::initializer_list<...> is in the stack of the current function, and after the function that uses the anonymous variable returns, ASAN marks the shadow memory of the anonymous variable as **f8** immediately.